magento/inventory
# Fix product cannot be added to wishlist on website assigned to custom stock

Unable to add product with multi-source inventory to wishlist with





### Description (*)


- Added generic "in-stock" filter for select object in CatalogInventory.
- Added plugin in InventoryCatalog to handle in-stock constraint for custom stock
- Added new module InventoryWishlist for MFTF test coverage of this scenario

The new generic "in-stock" filter can now be used to filter products or any product related entities. As for now, existing filters are restricted to product collections leading to code duplication for other entities. For instance:
\Magento\InventoryCatalog\Model\ResourceModel\AddIsInStockFieldToCollection
\Magento\InventoryCatalog\Model\ResourceModel\AddIsInStockFilterToCollection
\Magento\InventoryCatalog\Plugin\CatalogInventory\Model\Indexer\ModifySelectInProductPriceIndexFilter

### Fixed Issues (if relevant)

1. Fixes magento/inventory#3018: Unable to add product with multi-source inventory to wishlist with



### Manual testing scenarios (*)


1. Create a new inventory source -named 'new-source' in the Magento Admin
2. Create a new stock inventory -named 'new-stock' in the Magento Admin, assign 'new-source' and default website to the 'new-stock'
3. Create a simple product and only assign "new-stock" as in inventory
4. Go to storefront and log in to a customer account
5. Visit the the simple product product detail page in frontend
6. Add the product to the wishlist

ACTUAL 🔴 
The product is not added to the wishlist. The following error message is shown: _We can't add the item to Wish List right now: Cannot add product without stock to wishlist._

EXPECTED 🟢 
- [ ] The product is added to the wishlist and successful message is shown.
- [ ] The product is visible in customer wishlist 

### Related Pull Requests
https://github.com/magento-mpi/magento2ce/pull/526
https://github.com/magento-mpi/magento2-infrastructure/pull/34
https://github.com/magento-commerce/magento2-infrastructure/pull/1307

### Questions or comments


### Contribution checklist (*)
